src/celengine/nebula.cpp

```diff
@@ -22,23 +23,47 @@
 namespace engine = celestia::engine;
 namespace util = celestia::util;
 
+using namespace std::string_view_literals;
 using util::GetLogger;
 
+struct NebulaTypeName
+{
+    const char* name;
+    NebulaType type;
+};
+
+constexpr std::array NebulaTypeNames =
+{
+    NebulaTypeName{ "Emission", NebulaType::Emission },
+    NebulaTypeName{ "Reflection",  NebulaType::Reflection },
+    NebulaTypeName{ "Dark",  NebulaType::Dark },
+    NebulaTypeName{ "Planetary",  NebulaType::Planetary },
+    NebulaTypeName{ "SupernovaRemnant",  NebulaType::SupernovaRemnant },
+    NebulaTypeName{ "HII_Region", NebulaType::HII_Region },
+    NebulaTypeName{ "Protoplanetary", NebulaType::Protoplanetary },
+    NebulaTypeName{ " ", NebulaType::NotDefined },
+};
+
 const char*
 Nebula::getType() const
 {
-    return "Nebula";
+    return NebulaTypeNames[static_cast<std::size_t>(type)].name;
 }
 
 void
-Nebula::setType(const std::string& /*typeStr*/)
+Nebula::setType(const std::string& typeStr)
 {
+    type = NebulaType::NotDefined;
+    auto iter = std::find_if(std::begin(NebulaTypeNames), std::end(NebulaTypeNames),
+                             [&](const NebulaTypeName& n) { return n.name == typeStr; });
+    if (iter != std::end(NebulaTypeNames))
+        type = iter->type;
 }
 
 std::string
 Nebula::getDescription() const
 {
-    return _("Nebula");
+    return fmt::sprintf(_("Nebula: %s"), getType());
```

Comment:
```suggestion
    return fmt::format(_("Nebula: {}"), getType());
```
Don't use C-style function